config/compilerFlags.cmake

```diff
@@ -5,6 +5,9 @@ if ( MINGW OR UNIX ) # MINGW, Linux, APPLE, CYGWIN
         set(COMPILER_IS_CLANG ON)
     endif()
 
+    set (CMAKE_CXX_FLAGS_RELEASE    "-O3 -DNDEBUG")

```

Comment:
`-O3` is not recommended for production use and we should not add that explicitly as it might have precedence over `-O2` that a packager would set via `CXXFLAGS`

But a similar config should be saved somewhere, as when I run cmake, I get the following final output:
```
-- ------------------------------------------------------------------
-- Compiler info: GNU (/usr/bin/g++) ; version: 7.2.1
-- Compiler flags
-- General:            -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W -std=gnu++98
-- Extra:               -Wextra -Wlogical-op -Wdouble-promotion -Wshadow -Wuseless-cast -Wpointer-arith -Wformat=2 -Warray-bounds=2 -Wduplicated-cond -Wduplicated-branches -Wrestrict
-- Debug:             -g
-- Release:           -O2 -DNDEBUG
-- RelWithDebInfo:    -O2 -g -DNDEBUG
-- MinSizeRel:        -Os -DNDEBUG
```
where Release is something comparable to your setting.

Reply:
I know I also prefers O2, but the value was defaulted to O3 in my machine ... Maybe I had that value cached by other IDE or something ... I will double check.

Reply:
Are you sure you are getting -O2 by default ? After running CMake in a clean build directory I get this: 

```
-- Compiler info: GNU (/usr/bin/c++) ; version: 7.2.0
-- Compiler flags
-- General:            -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W -std=gnu++98
-- Debug:             -g
-- Release:           -O3 -DNDEBUG
-- RelWithDebInfo:    -O2 -g -DNDEBUG
-- MinSizeRel:        -Os -DNDEBUG
```

Reply:
hummm it seems that different distributions are patching CMake to set different defaults ... I think the best I can do is not to set these values for `CMAKE_CXX_FLAGS_RELEASE` but just for `CMAKE_CXX_FLAGS_DEBUG`

Reply:
Interesting, I did not know that there were differences in the default flags.